### PR TITLE
Update Freshli Jupyter Documentation

### PR DESCRIPTION
**Pull request by dan-hein**
### Introduction
While working on Freshli Issue #30 it became a necessity to setup dotnet-interactive and Jupyter to run some statements within a Jupyter notebook. The initial setup for this came with a few issues, and once setup, the examples provider required additional attention and fixes to get the desired output. To help future hands on the project that need to do the same setup and testing, I made changes to the Jupyter Notebook section of the Freshli README. My goal with these changes are to provide a clear path for someone who might not have worked with Jupyter before. I believe further detail can be added if necessary, but since there are plenty of resources that show how to interact with Jupyter once it is running, I decided not to add those instructions at this time.

### Changes
The first item that I wanted clear up was the installation of Jupyter and dotnet-interactive. Dotnet has an older platform that it used to use, and no longer supports, and the documentation online is a bit muddled. To remedy this, I added a link to what I found are the most update to date steps from the official source. I also added a link for the Jupyter installation for first time users.

Next, I broke the documentation down into sections to help aide the flow, as without the previous steps, continuing forward would cause issues.

The sample notebook attached has some additional steps that require a specific environment setup. Using this sample is also no longer needed with the instructions provided below, so I noted both of these things before moving on to the Jupyter commands.

The code provided has not been maintained as Freshli has changed, so there were some adjustments necessary to make this functional. 

### Closing
If there is anything else needed, or if my documentation is unclear, please let me know and I'd be more than happy to take care of it and clarify!

### CODE REVIEW

**Comment by mrbiggred** [comment]
@dan-hein I'll review this one and in the process setup Jupyter Notebooks.  I've never used Jupyter Notebooks before so I think I'll make a good documentation tester.

File: README.md [hunk 1]
```diff
@@ -212,16 +212,23 @@ The console output is designed to make it easy to copy and paste into a spreadsh
...
 
-1. Install Jupyter-Lab or nteract
-1. Install dontet/interactive
+1. Install [Jupyter-Lab](https://jupyterlab.readthedocs.io/en/stable/getting_started/installation.html) or nteract.
+1. Install [dontet/interactive.](https://github.com/dotnet/interactive/blob/main/docs/NotebooksLocalExperience.md)
 
+#### Sample Notebook
 A [sample notebook](https://github.com/corgibytes/freshli/blob/main/Sample.ipynb) can be found in this repository.
+To run the statements within this sample file, you will need to pull down a local version of
+[XPlot]() and run the build script located inside of this project. Note that, currently,
```
Comment (mrbiggred) [inline README.md:224]: I think you are missing the link to XPlot.  Do we need specific version?

File: README.md [hunk 2]
```diff
@@ -212,16 +212,23 @@ The console output is designed to make it easy to copy and paste into a spreadsh
...
+the XPlot build will fail unless the 3.1.201 SDK is installed on your local machine as a breaking
+change to this project was introduced in dotnet SDK 3.1.300. However, for the sample below, this step is unnecessary.
```
[the remark this answers is not in the document]
Comment (mrbiggred) [reply README.md:226]: Never mind I just found:

https://github.com/dotnet/interactive/issues/534

It appears that Jupyter ships with it's own version of XPlot.  Lots of hoops to jump through to get the sample [sample notebook](https://github.com/corgibytes/freshli/blob/main/Sample.ipynb) to work.  Is there anything we can do to make it simpler?  Something that does not involve compiling your own version of `dotnet/interactive`.

Should we remove the custom XPlot features, like `stackgroup` form the sample for now?  Then once @mscottford submits his PR we could look at re-adding the `stackgroup` feature to the sample.

Thoughts?

**Review by mrbiggred — CHANGES_REQUESTED** [review]
Details in the code comments but my big question is if we should update the Sample.ipynb file so it works without the need to re-compile `dotnet/interactive`.  At least that is what I think you need to do to get the sample to run.

I was thinking of asking you to put the simple sample in the ReadMe into a `*.jpynb` file but I'm not sure that is a good idea.  It appears the `*.jpynb` don't play well with version control.

Thoughts?

File: README.md [hunk 3]
```diff
@@ -241,14 +249,12 @@ using Freshli.Languages.Ruby;
 using Freshli.Languages.Php;
 using Freshli.Languages.Python;
 
-ManifestFinder.Register<RubyBundlerManifestFinder>();
-ManifestFinder.Register<PhpComposerManifestFinder>();
-ManifestFinder.Register<PipRequirementsTxtManifestFinder>();
+ManifestFinder.RegisterAll();
```
[the remark this answers is not in the document]
Comment (dan-hein) [reply README.md:252]: Ah I missed that one, can do!

File: README.md (hunk 1 above)
Comment (dan-hein) [reply README.md:224]: The version of Xplot used in the Sample is the version that is in the Corgibytes Repo

Comment (dan-hein) [reply README.md:224]: I can add a link

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dan-hein) [reply README.md:226]: That might be a good way forward, though I wondering if we even need this sample with the other documentation provided.

[the remark this answers is not in the document]
Comment (dan-hein) [reply README.md:226]: @mrbiggred Do you think that the example in the README is sufficient?

[the remark this answers is not in the document]
Comment (mrbiggred) [reply README.md:226]: @dan-hein I think the example in the README is a good starting point.  It is a good smoke test that you have Jupyter and Freshli setup correctly.  It would be nice to have both a simple smoke test example and a more complicated example.

The issue I have with the current [Sample](https://github.com/corgibytes/freshli/blob/main/Sample.ipynb) is that it does not work out of the box.  At least not for me.  I would like to keep this more complex example but remove the `stackgroup` option at the end that requires a customer XPlot.

```
var areaSeries = new List<Graph.Scatter>();

foreach (var name  in names) {
    areaSeries.Add(new Graph.Scatter {
        name = name,
        x = project.Select(r => r.Date),
        y = project.Where(r => r.LibYear.Any(p => p.Name == name)).Select(r => r.LibYear[name].Value),
        fill = "tonexty",
        stackgroup = "one"  # REMOVE THIS LINE
    });
}

var chart = Chart.Plot(areaSeries.ToArray());
chart.WithTitle("LibYear over time per dependency");
display(chart)
```

Finally I think it would be better if the simple smoke test example is `ipynb` file instead of copy/paste from the readme.  Thoughts?

File: README.md (hunk 1 above)
Comment (mrbiggred) [reply README.md:224]: If we remove the customer XPlot requirement than I think we can remove the documentation about install XPlot.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrbiggred) [reply README.md]: @dan-hein I like @mscottford's idea of setting up a Docker container but not right now.  I would like to see these ReadMe changes merged so @dfar-io can do the [v0.3.0](https://github.com/corgibytes/freshli/milestone/4) ReadMe clean up (#242).

For now I think it's best if the Jupyter instructions said something like:

1) Install Juptyer with link to install instructions and leave it up to the reader to install however they want.
2) Open the smoke test sample in Jupyter.
3) Change the link to the Freshli DLL.
4) Run the sample.

We can also have a link to the more complicated Jupyter sample but that sample should have the `stackgroup` property removed for now.  Once fslaborg/XPlot#144 is merged we can update the ReadMe.  We can also update the samples when we push Freshli to NuGet.

In summary update the documentation and samples to work with what we have now.  We will update the documentation when Jupyter/XPlot PRs are merged and when Freshli Core makes it opening debut on NuGet.

Thoughts?

**Review by mrbiggred — CHANGES_REQUESTED** [review]
@dan-hein see my [comments](https://github.com/corgibytes/freshli/pull/226#discussion_r581258592) with @mscottford discussion for more details but in summary I would like to see:

1) Two sample `ipynb` files, one for a smoke test and one to show off all the Freshli features, named something appropriate.
2) Remove the smoke test example from the ReadMe since it's in it's own file.

Let me know what you think.
